tonic-build/src/client.rs

```diff
@@ -99,12 +99,14 @@ fn generate_unary(method: &Method, proto: &str, path: String) -> TokenStream {
     let (request, response) = crate::replace_wellknown(proto, &method);
 
     quote! {
-        pub async fn #ident(&mut self, request: tonic::Request<#request>)
-            -> Result<tonic::Response<#response>, tonic::Status> {
+        pub async fn #ident<R: tonic::IntoRequest<Message = #request>> (
+            &mut self,
+            request: R,
```

Comment:
Rather than using the `R` type parameter, this could just be
```rust
pub async fn #ident(&mut self, request: impl tonic::IntoRequest<Message = #request>)
     -> ...
```